tests/testthat/test-add_gloabl_p.tbl_uvregression.R

```diff
@@ -0,0 +1,156 @@
+      (car::Anova(lm(marker ~ age, trial)))$`Pr(>F)`[1],
+      (car::Anova(lm(marker ~ grade, trial)))$`Pr(>F)`[1]
+    )
+  )
+})
+
+test_that("add_global_p.tbl_uvregression(include)", {
+  tbl <- trial |>
+    tbl_uvregression(method = lm, y = marker, include = c("age", "grade"))
+
+  res1 <- tbl |> add_global_p(include = age)
+
+  # one expected p-value for age
+  expect_equal(
+    sum(!is.na(res1$table_body$p.value)),
+    1
+  )
+})
+
+test_that("add_global_p.tbl_uvregression(type)", {
+  tbl <- trial |>
+    tbl_uvregression(method = lm, y = marker, include = c("age", "grade"))
+
+  res2 <- tbl |> add_global_p(type = "II")
+
+  # 4 expected p-values, 2 for each variable, 2 for each level of grade
+  expect_equal(
+    sum(!is.na(res2$table_body$p.value)),
+    2
+  )
+})
+
+test_that("add_global_p.tbl_uvregression(keep)", {
+  tbl <- trial |>
+    tbl_uvregression(method = lm, y = marker, include = c("age", "grade"))
+
+  res3 <- tbl |> add_global_p(keep = TRUE)
+
+  # 4 expected p-values, 2 for each variable, 2 for each level of grade
+  expect_equal(
+    sum(!is.na(res3$table_body$p.value)),
+    4
+  )
+})
+
+test_that("add_global_p.tbl_uvregression(anova_fun)", {
+  tbl <- trial |>
+    tbl_uvregression(method = lm, y = marker, include = c("age", "grade"))
+
+  res4 <- tbl |> add_global_p(anova_fun = cardx::ard_aod_wald_test)
+
+  # two model terms, two p values
+  expect_equal(
+    sum(!is.na(res4$table_body$p.value)),
+    2
+  )
+
+  # p-values match when using aod_wald_test
+  expect_equal(
+    res4$table_body$p.value[1:2],
+    c(
+      lm(marker ~ age, trial) |>
+        cardx::ard_aod_wald_test() |>
+        dplyr::filter(variable %in% c("age", "grade") & stat_name == "p.value") |>
+        dplyr::pull(stat) %>%
+        unlist(),
+      lm(marker ~ grade, trial) |>
+        cardx::ard_aod_wald_test() |>
+        dplyr::filter(variable %in% c("age", "grade") & stat_name == "p.value") |>
+        dplyr::pull(stat) %>%
+        unlist()
+    )
+  )
+})
+
+test_that("add_global_p.tbl_uvregression(anova_fun) inappropriate anova function", {
+  tbl <- trial |>
+    tbl_uvregression(method = lm, y = marker, include = c("age", "grade"))
+
+  broken_anova <- function(x) {
+    x + 1
+  }
+
+  expect_error(
+    res5 <- tbl |> add_global_p(anova_fun = broken_anova),
+    regexp = "There was an error running `anova_fun`"
+  )
+})
+
+test_that("geeglm model for add_global_p.tbl_uvregression()", {
+  res5 <- respiratory |>
+      tbl_uvregression(
+        method = geepack::geeglm,
+        y = outcome,
+        include = c("treat", "baseline"),
+        method.args = list(
+          family = binomial,
+          id = id,
+          corstr = "exchangeable"
+        ),
+      ) |>
+      add_global_p()
+
+  gee_model <- geeglm(
+    outcome ~ treat + baseline,
+    id = id,
+    data = respiratory,
+    family = binomial,
+    corstr = "exchangeable"
+  )
+  coef <- coef(gee_model)
+  vcov_matrix <- vcov(gee_model)
+
+
+  # not equal
```

Comment:
@ddsjoberg lmk if I'm doing something wrong but I'm not getting equal terms here, verifying the use of aod wald test for a gee model (also swapped out for a simpler example)

Reply:
Let's update to something like this:

```r
res5 <- geepack::respiratory |>
  tbl_uvregression(
    method = geepack::geeglm,
    y = outcome,
    include = c("treat", "baseline"),
    method.args = list(
      family = binomial,
      id = id,
      corstr = "exchangeable"
    ),
  ) |>
  add_global_p()

expect_equal(
  res5$table_body |> dplyr::filter(variable == "treat") |> dplyr::pull("p.value") |> getElement(1L),
  geepack::geeglm(
    outcome ~ treat, 
    geepack::respiratory, family = binomial,
    id = id,
    corstr = "exchangeable"
  ) |> 
    cardx::ard_aod_wald_test() |> # calculate Wald p-value
    dplyr::filter(variable == "treat", stat_name == "p.value") |> 
    dplyr::pull("stat") |> 
    unlist()
)

expect_equal(
  res5$table_body |> dplyr::filter(variable == "baseline") |> dplyr::pull("p.value") |> getElement(1L),
  geepack::geeglm(
    outcome ~ baseline, 
    geepack::respiratory, family = binomial,
    id = id,
    corstr = "exchangeable"
  ) |> 
    cardx::ard_aod_wald_test() |> # calculate Wald p-value
    dplyr::filter(variable == "baseline", stat_name == "p.value") |> 
    dplyr::pull("stat") |> 
    unlist()
)
```